src/models/decoder_only_pipeline.cpp

```diff
@@ -179,16 +204,31 @@ RoamingArray<float> DecoderOnlyPipelineState::Run(int current_length, RoamingArr
     // Run the intermediate pipeline state
     pipeline_state->Run(current_length, next_tokens, next_indices);
 
-    // Store the non-managed outputs from the current pipeline state in the ortvalue pool.
+    // Transfer ownership of all the non-managed outputs from the current pipeline state to the ortvalue store.
     // All non managed outputs are assumed to be on CPU
     for (size_t i = 0; i < pipeline_state->output_names_.size(); ++i) {
       if (std::none_of(output_names_.begin(), output_names_.end(),
+                       [&](const std::string& elem) { return elem == pipeline_state->output_names_[i]; }) &&
+          std::none_of(input_names_.begin(), input_names_.end(),
```

Comment:
just for my understanding, why was it changed from `output_names` to `input_names` here? also, what is the difference between a managed and non-managed output?

Reply:
Yes. managed inputs/outputs are the model inputs/outputs that ort-genai is aware of. For example, the kv cache, the input_ids, the position_ids and so on. The input/outputs are updated by ort-genai after each run (as required).

Non managed inputs and outputs are those pipeline-model inputs and outputs that ort-genai does not care/know of. For example intermediate outputs of one pipeline model that feed directly into the next pipeline model is a non managed input/output. These ortvalues are stored in the `ortvalue_store_` map.